### DISCUSSION TITLE
Large file storage with IPFS

### DISCUSSION THREAD

**Discussion by wwwehr** [NillionNetwork/discussions#147] [Show and tell]
### Show and tell project type

Community Builder

### Github Repo Link

https://github.com/wwwehr/nillion-ipfs-store

### Project Description

This is a python reference implementation of securely sending large files to decentralized storage. Encrypted files go on IPFS, and the decryption keys go to Nillion. Includes both store and retrieval.

### What problems does your project solve? How does it preserve privacy for users?

- Problem 1: large files (1GB+) are not supported in native Nillion at the moment
- Problem 2: large file encryption schemes in python are a challenge (fernet requires file in memory)

### How does the project use Nillion? Describe and link to any Nada programs

Nillion is used to store the information needed to reconstruct the encrypted file, including the IPFS CID.

**Comment by kkk2791** [comment]
Have you solved the problem?

**Comment by litang1028** [comment]
我要好好研究这个项目
